src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/SimpleConfigModel.cs

```diff
@@ -189,52 +189,130 @@ IReadOnlyList<FileSource> IRunnableProjectConfig.Sources
+                    defaultSpecials.Add(new SpecialOperationConfigParams("**/*.asp", "<!--", ConditionalType.Xml));
+                    defaultSpecials.Add(new SpecialOperationConfigParams("**/*.aspx", "<!--", ConditionalType.Xml));
+
+                    List<KeyValuePair<string, IGlobalRunConfig>> specialOperationConfig = new List<KeyValuePair<string, IGlobalRunConfig>>();
+
+                    // put the custom configs first in the list
+                    HashSet<string> processedGlobs = new HashSet<string>();
+
+                    foreach (ICustomFileGlobModel customGlobModel in SpecialCustomSetup)
+                    {
+                        SpecialOperationConfigParams defaultParams = defaultSpecials.Where(x => x.Glob == customGlobModel.Glob).FirstOrDefault();
+                        IGlobalRunConfig runConfig;
+
+                        if (defaultParams != null)
                         {
-                            ["**/*.json"] = ProduceOperationSetup("//", false, ConditionalType.CWithComments),
-                            ["**/*.css"] = ProduceOperationSetup("/*", false, ConditionalType.CBlockComments),
-                            ["**/*.css.min"] = ProduceOperationSetup("/*", false, ConditionalType.CBlockComments),
-                            ["**/*.cs"] = ProduceOperationSetup("//", false, ConditionalType.CNoComments),
-                            ["**/*.cpp"] = ProduceOperationSetup("//", false, ConditionalType.CNoComments),
-                            ["**/*.hpp"] = ProduceOperationSetup("//", false, ConditionalType.CNoComments),
-                            ["**/*.h"] = ProduceOperationSetup("//", false, ConditionalType.CNoComments),
-                            ["**/*.*proj"] = ProduceOperationSetup("<!--/", false, ConditionalType.Xml),
-                            ["**/*.*html"] = ProduceOperationSetup("<!--", false, ConditionalType.Xml),
-                            ["**/*.*htm"] = ProduceOperationSetup("<!--", false, ConditionalType.Xml),
-                            ["**/*.jsp"] = ProduceOperationSetup("<!--", false, ConditionalType.Xml),
-                            ["**/*.asp"] = ProduceOperationSetup("<!--", false, ConditionalType.Xml),
-                            ["**/*.aspx"] = ProduceOperationSetup("<!--", false, ConditionalType.Xml),
-                        };
-                    _specialOperationConfig = operationSpecials;
+                            // the custom config is for a default-defined glob. Producing the setup uses the custom info.
+                            runConfig = ProduceOperationSetup(defaultParams.SwitchPrefix, false, defaultParams.ConditionalStyle, customGlobModel);
+                        }
+                        else
+                        {
+                            // the custom config is not for a known glob type. Only use the custom info
+                            runConfig = ProduceOperationSetup(string.Empty, false, ConditionalType.None, customGlobModel);
```

Comment:
The switch prefix will have to come from the config so that we don't break the language of whatever the custom pattern matches